ui/app/AppLayouts/Profile/views/AdvancedView.qml

```diff
@@ -386,34 +386,59 @@ ScrollView {
                 }
             }
 
-            // TODO: replace with StatusQ component
+            StatusSectionHeadline {
+                text: qsTr("Developer features")
+                topPadding: Style.current.bigPadding
+                bottomPadding: Style.current.padding
+            }
+
+            Separator {
+                anchors.topMargin: Style.current.bigPadding
+                anchors.left: parent.left
+                anchors.leftMargin: -Style.current.padding
+                anchors.right: parent.right
+                anchors.rightMargin: -Style.current.padding
+            }
+
             StatusSettingsLineButton {
-                text: qsTr("Download messages")
+                text: qsTr("Full developer mode")
                 isSwitch: true
-                switchChecked: localAccountSensitiveSettings.downloadChannelMessagesEnabled
+                visible: {
+                    return !localAccountSensitiveSettings.downloadChannelMessagesEnabled ||
+                        !root.store.profileModuleInst.isTelemetryEnabled ||
+                        !root.store.profileModuleInst.isDebugEnabled ||
+                        !root.store.profileModuleInst.isAutoMessageEnabled
+                }
                 onClicked: {
-                    localAccountSensitiveSettings.downloadChannelMessagesEnabled = !localAccountSensitiveSettings.downloadChannelMessagesEnabled
+                    openPopup(enableDeveloperFeaturesConfirmationDialogComponent)
```

Comment:
This is a switch, but it seems to act as a button. Either we can change it to a button, or rename and change the functions to act as toggle instead. Because right now, you can enable, but never disable the full dev mode

Reply:
Well spotted, it can only enabled. For now it is only visible if not all the items are active via:
```
visible: {
                    return !localAccountSensitiveSettings.downloadChannelMessagesEnabled ||
                        !root.store.profileModuleInst.isTelemetryEnabled ||
                        !root.store.profileModuleInst.isDebugEnabled ||
                        !root.store.profileModuleInst.isAutoMessageEnabled
                }
```

So you would prefer a button that is disable if all items are enabled right?

Reply:
I did miss the `visible` property. But yeah, either make it a button (just removing `isSwitch` works I think, or make it possible to uncheck this to set all of those props to `false`

Reply:
Gotcha, updated! and thanks for the tip